doc/developers/develop.rst

```diff
@@ -518,9 +518,15 @@ _skip_test (default=False)
     *very good* reason.
 
 _xfail_checks (default=False)
-    dictionary ``{check_name : reason}`` of common checks to mark as a
-    known failure, with the associated reason. Don't use this unless you have a
-    *very good* reason.
+    dictionary ``{check_name: reason}`` of common checks that will be marked
+    as `XFAIL` for pytest, when using
+    :func:`~sklearn.utils.estimator_checks.parametrize_with_checks`. This tag
+    currently has no effect on
+    :func:`~sklearn.utils.estimator_checks.check_estimator`.
+    Of all tags, the usage of this one is highly subject to change because we
+    are trying to make it more flexible in the future. Don't use this unless
+    you have a *very good* reason, and be prepared for breaking changes in
+    the future.
```

Comment:
```suggestion
    Warning: the API of this tag is likely to change in v0.24 in a backward compatible way.
```

I don't agree with recommending not to use it. Yes, the API may change in the future (but we are not even 100% sure), stating it and letting users decide if they are willing to make the migration for v0.24 is enough IMO.

Reply:
OK I see "Don't use this unless you have a *very good* reason." was there before, we can leave it then.

Reply:
Though yeah the "don't use this" is more about "be careful about diverting from strict compatibility". I'll reorder